pkg/api/methodHandlers.go

```diff
@@ -131,6 +138,38 @@ func (h *methodHandlers) execute(ctx context.Context, tracker attribute.Tracker,
 	return newStatus(code.Code_OK)
 }
 
+// execWrapper is responsible for dispatching calls to the manager and recovering from panics in adapter impls.
+// It also monitors the RPC's deadline for timeouts.
+func (h *methodHandlers) execWrapper(ctx context.Context, conf *aspectsupport.CombinedConfig, ab attribute.Bag) (*aspectsupport.Output, error) {
+	type result struct {
+		out *aspectsupport.Output
+		err error
+	}
+	c := make(chan result)
+	// TODO: should we use some threadpool impl instead of just spawning a thread?
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				glog.Warningf("Recovered from panic in adapter '%s'", conf.Adapter.Name)
```

Comment:
Should log this as an error.

Should disable/remove the aspect once it has failed so we don't end up continuously failing. Maybe we should keep a counter and allow N failures per minute or something....